relay-general/src/pii/config.rs

```diff
@@ -8,10 +8,33 @@ use serde::{de::Error, Deserialize, Deserializer, Serialize, Serializer};
 
 use crate::pii::{CompiledPiiConfig, Redaction};
 use crate::processor::SelectorSpec;
+use failure::Fail;
+
+#[derive(Clone, Debug, Fail)]
+pub enum PiiConfigError {
+    #[fail(display = "could not parse regex")]
+    RegexError(#[cause] regex::Error),
+}
 
 /// A regex pattern for text replacement.
 #[derive(Clone)]
-pub struct Pattern(pub Regex);
+pub struct Pattern(Regex);
+
+impl Pattern {
+    pub fn new(s: &str, case_insensitive: bool) -> Result<Self, PiiConfigError> {
```

Comment:
_nit_: By convention, `new` does not return results. You could name this:

```suggestion
    pub fn parse(s: &str, case_insensitive: bool) -> Result<Self, PiiConfigError> {
```